cmd/juju/controller/listmodels_test.go

```diff
@@ -77,41 +77,78 @@ func (f *fakeModelMgrAPIClient) AllModels() ([]base.UserModel, error) {
 	return f.convertInfosToUserModels(), nil
 }
 
-func (f *fakeModelMgrAPIClient) ListModelSummaries(user names.UserTag) ([]params.ModelSummaryResult, error) {
-	f.MethodCall(f, "ListModelSummaries", user)
+func (f *fakeModelMgrAPIClient) ListModelSummaries(user string, all bool) ([]base.UserModelSummary, error) {
+	f.MethodCall(f, "ListModelSummaries", user, all)
 	if f.err != nil {
 		return nil, f.err
 	}
-	results := make([]params.ModelSummaryResult, len(f.infos))
+	results := make([]base.UserModelSummary, len(f.infos))
 	for i, info := range f.infos {
-		results[i] = params.ModelSummaryResult{}
+		results[i] = base.UserModelSummary{}
 		if info.Error != nil {
 			results[i].Error = info.Error
 			continue
 		}
-		results[i].Result = &params.ModelSummary{
-			Name:               info.Result.Name,
-			UUID:               info.Result.UUID,
-			ControllerUUID:     info.Result.ControllerUUID,
-			ProviderType:       info.Result.ProviderType,
-			DefaultSeries:      info.Result.DefaultSeries,
-			CloudTag:           info.Result.CloudTag,
-			CloudRegion:        info.Result.CloudRegion,
-			CloudCredentialTag: info.Result.CloudCredentialTag,
-			OwnerTag:           info.Result.OwnerTag,
-			Life:               info.Result.Life,
-			Status:             info.Result.Status,
-			Migration:          info.Result.Migration,
-			SLA:                info.Result.SLA,
-			AgentVersion:       info.Result.AgentVersion,
+		cloud, err := names.ParseCloudTag(info.Result.CloudTag)
+		if err != nil {
+			cloud = names.NewCloudTag("aws")
+
```

Comment:
Considering this is a mock, I'd rather not pass c into here and Assert outside of this call.. But I am not convinced that even that is necessary. This whole logic was useful when I was transforming existing tests - some 'tags" were not actually valid and we have never discovered them because  no code path exercised by the old tests checked these tags. I've kept the code here in case we ever need to modify unit tests and will try to setup invalid test data...